test/viewmodels/users_ranking_view_model_test.dart

```diff
@@ -0,0 +1,134 @@
+
+      expect(
+        ranking.rankElementDetailsList.length,
+        UsersRankingViewModel.rankingLimit,
+      );
+    });
+
+    test("Logged in user has correct values", () async {
+      final ranking =
+          await rankingContainer.read(usersRankingViewModelProvider.future);
+      final currentUserRanking = ranking.userRankElementDetails;
+
+      expect(
+        currentUserRanking.userDisplayName,
+        testingUserData.displayName,
+      );
+      expect(
+        currentUserRanking.userUserName,
+        testingUserData.username,
+      );
+      expect(
+        currentUserRanking.centauriPoints,
+        testingUserData.centauriPoints,
+      );
+      expect(
+        currentUserRanking.userRank,
+        isNull,
+      );
+    });
+
+    test("View model returns a descending order with correct ranks", () async {
+      final ranking =
+          await rankingContainer.read(usersRankingViewModelProvider.future);
+      final leaderboard = ranking.rankElementDetailsList;
+
+      // Check that all ranks are non null and correct
+      final lastRank =
+          leaderboard.map((e) => e.userRank!).reduce((value, element) {
+        expect(value + 1, element);
+        return element;
+      });
+      expect(lastRank, UsersRankingViewModel.rankingLimit);
+
+      // Check that all centauri points are in descending order
+      final lastCentauri =
+          leaderboard.map((e) => e.centauriPoints).reduce((value, element) {
+        expect(value >= element, isTrue);
+        return element;
+      });
+      expect(lastCentauri, UsersRankingViewModel.rankingLimit);
+    });
+
+    test("View model refreshes correctly", () async {
+      final ranking =
+          await rankingContainer.read(usersRankingViewModelProvider.future);
+      final topUser = ranking.rankElementDetailsList.first;
+
+      // Create a user with more points than current top user
+      final newTopUserFireStore = UserFirestore(
+        uid: const UserIdFirestore(value: "my_top_user_uid"),
+        data: UserData(
+          username: "top user",
+          displayName: "top user",
+          joinTime: Timestamp.now(),
+          centauriPoints: topUser.centauriPoints + 1,
+        ),
+      );
+
+      await setUserFirestore(
+        fakeFirestore,
+        newTopUserFireStore,
+      );
+
+      // Refresh the viewmodel
+      await rankingContainer
+          .read(usersRankingViewModelProvider.notifier)
+          .refresh();
+      // delay for async execution of refresh
+      await Future.delayed(Durations.short1);
+
+      // Retrieve the new rankings from viewmodel
+      final newRanking =
+          await rankingContainer.read(usersRankingViewModelProvider.future);
+      final newTopUser = newRanking.rankElementDetailsList.first;
+
+      // Check that the new top user was correctly update
+      expect(
+        newTopUser.userUserName != topUser.userUserName,
+        isTrue,
+      );
```

Comment:
There's also a matcher for this. I think this makes code more readable, and more homogeneous, and, since we use matchers extensively in the rest of the codebase, they should also be used extensively here:
```suggestion
      expect(newTopUser.userUserName, isNot(equals(topUser.userUserName)));
```

Reply:
I don't think it is more readable, but sure... done in d2a1983 (nitpick)